src/datasources/db/cached-query-resolver.spec.ts

```diff
@@ -0,0 +1,74 @@
+import { fakeJson } from '@/__tests__/faker';
+import { FakeCacheService } from '@/datasources/cache/__tests__/fake.cache.service';
+import { CachedQueryResolver } from '@/datasources/db/cached-query-resolver';
+import { ILoggingService } from '@/logging/logging.interface';
+import { faker } from '@faker-js/faker';
+import postgres, { MaybeRow } from 'postgres';
+
+const mockLoggingService = jest.mocked({
+  debug: jest.fn(),
+} as jest.MockedObjectDeep<ILoggingService>);
+
+const mockQuery = jest.mocked({ catch: jest.fn() } as jest.MockedObjectDeep<
+  postgres.PendingQuery<MaybeRow[]>
+>);
+
+describe('CachedQueryResolver', () => {
+  let fakeCacheService: FakeCacheService;
+  let target: CachedQueryResolver;
+
+  beforeAll(() => {
+    fakeCacheService = new FakeCacheService();
+    target = new CachedQueryResolver(mockLoggingService, fakeCacheService);
+  });
+
+  afterEach(() => {
+    jest.clearAllMocks();
+    fakeCacheService.clear();
+  });
+
+  describe('get', () => {
+    it('should return the content from cache if it exists', async () => {
+      const cacheDir = { key: 'key', field: 'field' };
+      const ttl = faker.number.int({ min: 1, max: 1000 });
+      const value = fakeJson();
+      await fakeCacheService.set(cacheDir, JSON.stringify(value), ttl);
+
+      const actual = await target.get({
+        cacheDir,
+        query: mockQuery,
+        ttl,
+      });
+
+      expect(actual).toBe(value);
+      expect(mockLoggingService.debug).toHaveBeenCalledWith({
+        type: 'cache_hit',
+        key: 'key',
+        field: 'field',
+      });
+      expect(mockQuery.catch).not.toHaveBeenCalled();
```

Comment:
I've changed the approach in [4f3b976](https://github.com/safe-global/safe-client-gateway/pull/1777/commits/4f3b976489406cc6a8c1771c0611645d36a3505c), to explicitly call `execute` on the query so the promise handling is simpler (and also the testing). According to [the library docs](https://github.com/porsager/postgres?tab=readme-ov-file#execute), it would cause the execution to happen on a single Nodejs tick.